src/Relude/Extra/Type.hs

```diff
@@ -114,3 +117,17 @@ Maybe (Snd (Int, Text)) :: *
 type family Snd (t :: k) :: k' where
     Snd '(_, y) = y
     Snd  (_, y) = y
+
+
+{- | Check that a type is an element of a list:
```

Comment:
Let's also add one more example with an empty list to show how it works on empty lists